src/keys.rs

```diff
@@ -1,23 +1,23 @@
 use crate::hasher::Hasher;
-use crate::params::{Params, WotsError, MAX_MSG_SIZE, SEED_SIZE};
+use crate::params::{ComputeLaddersMode, Params, WotsError, MAX_MSG_SIZE, SEED_SIZE};
 
 #[cfg(feature = "std")]
 use rand::{rngs::OsRng, RngCore};
 
 /// Size of WOTS+ public keys
 pub const PK_SIZE: usize = 32;
 
-pub struct Key<PRFH: Hasher, MSGH: Hasher> {
-    p_seed: [u8; SEED_SIZE],
-    chains: Option<Vec<Vec<u8>>>,
+pub struct Key<PRFH: Hasher + Clone, MSGH: Hasher + Clone> {
+    pub p_seed: [u8; SEED_SIZE],
+    pub chains: Option<Vec<Vec<u8>>>,
     secret_key: Vec<u8>,
     public_key: Option<Vec<u8>>,
```

Comment:
Is it possible to make the `public_key` and `secret_key` `pub` also, see https://github.com/dalek-cryptography/ed25519-dalek/blob/ad461f4f0f56507a2945510dedb914b3d3547c40/src/keypair.rs#L36

really need to get these two keys from `Key` in my substrate implementation

Reply:
yes, updated!